pkg/karmadactl/cmdinit/config/config_test.go

```diff
@@ -0,0 +1,307 @@
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const testConfig = `
+apiVersion: config.karmada.io/v1alpha1
+kind: KarmadaInitConfig
+metadata:
+ name: karmada-init
+spec:
+ certificates:
+   caCertFile: "/etc/karmada/pki/ca.crt"
+   caKeyFile: "/etc/karmada/pki/ca.key"
+   externalDNS:
+     - "localhost"
+     - "example.com"
+   externalIP:
+     - "192.168.1.2"
+     - "172.16.1.2"
+   validityPeriod: "8760h0m0s"
+ etcd:
+   local:
+     Repository: "registry.k8s.io/etcd"
+     Tag: "latest"
+     dataPath: "/var/lib/karmada-etcd"
+     initImage:
+       repository: "alpine"
+       tag: "3.19.1"
+     nodeSelectorLabels:
+       karmada.io/etcd: "true"
+     pvcSize: "5Gi"
+     replicas: 3
+     storageClassesName: "fast"
+     storageMode: "PVC"
+   external:
+     endpoints:
+       - "https://example.com:8443"
+     caFile: "/path/to/your/ca.crt"
+     certFile: "/path/to/your/cert.crt"
+     keyFile: "/path/to/your/key.key"
+     keyPrefix: "ext-"
+ hostCluster:
+   apiEndpoint: "https://kubernetes.example.com"
+   kubeconfig: "/root/.kube/config"
+   context: "karmada-host"
+   domain: "cluster.local"
+ images:
+   imagePullPolicy: "IfNotPresent"
+   imagePullSecrets:
+     - "PullSecret1"
+     - "PullSecret2"
+   kubeImageMirrorCountry: "cn"
+   kubeImageRegistry: "registry.cn-hangzhou.aliyuncs.com/google_containers"
+   kubeImageTag: "v1.29.6"
+   privateRegistry:
+     registry: "my.private.registry"
+ components:
+   karmadaAPIServer:
+     repository: "karmada/kube-apiserver"
+     tag: "v1.29.6"
+     replicas: 1
+     advertiseAddress: "192.168.1.100"
+     serviceType: "NodePort"
+     networking:
+       namespace: "karmada-system"
+       port: 32443
+   karmadaAggregatedAPIServer:
+     repository: "karmada/karmada-aggregated-apiserver"
+     tag: "v0.0.0-master"
+     replicas: 1
+   kubeControllerManager:
+     repository: "karmada/kube-controller-manager"
+     tag: "v1.29.6"
+     replicas: 1
+   karmadaControllerManager:
+     repository: "karmada/karmada-controller-manager"
+     tag: "v0.0.0-master"
+     replicas: 1
+   karmadaScheduler:
+     repository: "karmada/karmada-scheduler"
+     tag: "v0.0.0-master"
+     replicas: 1
+   karmadaWebhook:
+     repository: "karmada/karmada-webhook"
+     tag: "v0.0.0-master"
+     replicas: 1
+ karmadaDataPath: "/etc/karmada"
+ karmadaPKIPath: "/etc/karmada/pki"
+ karmadaCRDs: "https://github.com/karmada-io/karmada/releases/download/test/crds.tar.gz"
+ waitComponentReadyTimeout: 120
+`
+
+const invalidTestConfig = `
+apiVersion: v1alpha1
+kind: KarmadaInitConfig
+metadata:
+  name: karmada-init
+spec:
+  waitComponentReadyTimeout: "invalid-int"
+`
+
+func TestLoadInitConfiguration(t *testing.T) {
+	expectedConfig := &KarmadaInitConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "KarmadaInitConfig",
+			APIVersion: "config.karmada.io/v1alpha1",
```

Comment:
What happens when I pass `config.karmada.io/v1`, maybe add some test for it.

Reply:
In this section, the YAML file's group, version, and kind will be checked to see if they match the current configuration file format. If they don't match, the deployment will not proceed, and an error will be returned. Unit tests for this part have already been added to pkg/karmadactl/cmdinit/config/config_test.go.